src/app.jsx

```diff
@@ -196,59 +253,83 @@ class Application extends React.Component {
-                    this.updateImagesAfterEvent();
-                    this.updateContainersAfterEvent();
+                    this.setState({ [system ? "systemServiceAvailable" : "userServiceAvailable"]: true });
+                    this.updateImagesAfterEvent(system);
+                    this.updateContainersAfterEvent(system);
                     utils.monitor("GetEvents", {},
                                   message => {
-                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events);
-                                  },
-                                  () => {
-                                      this.setState({ serviceAvailable: false });
+                                      message.parameters && message.parameters.events && this.handleEvent(message.parameters.events, system);
+                                  }, isSystem => {
+                                      this.setState({ [isSystem ? "systemServiceAvailable" : "userServiceAvailable"]: false
+                                      });
                                   },
+                                  system
                     );
                 })
                 .catch(error => {
-                    if (error.name === "ConnectionClosed")
-                        this.setState({ serviceAvailable: false });
-                    else
+                    if (error.name === "ConnectionClosed") {
+                        this.setState({ [system ? "systemServiceAvailable" : "userServiceAvailable"]: false,
+                                        [system ? "systemContainersLoaded" : "userContainersLoaded"]: true,
+                                        [system ? "systemImagesLoaded" : "userImagesLoaded"]: true
+                        });
+                    } else
                         console.error("Failed to call GetVersion():", error);
                 });
     }
 
     componentDidMount() {
-        this.init();
+        this.checkUserService();
+        this.init(true);
+        cockpit.script("echo $XDG_RUNTIME_DIR")
+                .done(xrd => {
+                    sessionStorage.setItem('XDG_RUNTIME_DIR', xrd.trim());
+                    this.init(false);
+                })
+                .fail(e => console.log("Could not read $XDG_RUNTIME_DIR: ", e.message));
+    }
+
+    checkUserService() {
+        let argv = ["systemctl", "--user", "is-enabled", "io.podman.socket"];
+
+        cockpit.spawn(argv, { environ: ["LC_ALL=C"], err: "out" })
+                .then(() => this.setState({ userServiceExists: true }))
+                .catch((_, response) => {
+                    if (response.trim() !== "disabled")
+                        this.setState({ userServiceExists: false });
+                    else
+                        this.setState({ userServiceExists: true });
+                });
     }
 
     startService(e) {
```

Comment:
It's rather odd that this function enables both the system and the user service, unconditionally. Shouldn't this also get a boolean flag like the others?

Reply:
This is also used when neither is running (and you get the screen with `Start podman` which should start both units). But it could at least check if one service is already not recognized as running.

Reply:
But this function is called in following cases:
1. Only one unit is present and that unit is not active. We need to check both
2. There are two units, and exactly one is running. In that case the one is no-op and the other starts the second unit.